static/js/components/dashboard/CourseRow.js

```diff
@@ -72,115 +63,39 @@ export default class CourseRow extends React.Component {
     return firstRun;
   }
 
-  renderRowColumns(run: CourseRun): Array<React$Element<*>> {
+  courseAction = (run: CourseRun, actionType: string): React$Element<*>|null => {
     const {
-      course,
       now,
       prices,
       financialAid,
       hasFinancialAid,
       openFinancialAidCalculator,
       addCourseEnrollment,
-      openCourseContactDialog,
       setEnrollSelectedCourseRun,
       setEnrollCourseDialogVisibility,
       checkout
     } = this.props;
 
-    let lastColumnSize, lastColumnClass;
-    let columns = [
-      <Cell col={6} key="1">
-        <CourseDescription
-          courseRun={run}
-          courseTitle={course.title}
-          hasContactEmail={course.has_contact_email}
-          openCourseContactDialog={
-            R.partial(openCourseContactDialog, [course, this.hasPaidForAnyCourseRun(course)])
-          }
-        />
-      </Cell>
-    ];
-
-    if (this.shouldDisplayGradeColumn(run)) {
-      columns.push(
-        <Cell col={3} key="2">
-          <CourseGrade courseRun={run} course={course} />
-        </Cell>
-      );
-      lastColumnSize = 3;
-    } else {
-      lastColumnSize = 6;
-      lastColumnClass = 'long-description';
+    if (actionType === COURSE_ACTION_ENROLL && !isEnrollableRun(run)) {
+      return null;
```

Comment:
This can happen in the case when the user is not yet enrolled in a course but there is not a future enrollable run. The `renderEnrollableCourseInfo` method (on this class) is called in the case where a user is not already enrolled in any course run for the course, and in that method we call `this.courseAction(run, COURSE_ACTION_ENROLL);`. If `run` is an enrollable run (meaning the dates line up correctly and so on) we show an enroll button, and if it is not enrollable we don't want to show the button.

Does that make sense?